front/src/modules/scenario/components/ScenarioExplorer/ScenarioExplorer.tsx

```diff
@@ -140,23 +161,28 @@ const ScenarioExplorer = ({
               <img src={scenarioIcon} alt="scenario icon" />
               <span className="scenario-explorator-card-head-legend">{t('scenarioLegend')}</span>
               <div className="scenario-explorator-card-head-scenario">
-                <span className="text-truncate" title={scenarioDetails.name}>
-                  {scenarioDetails.name}
+                <span className="text-truncate" title={scenario?.name}>
+                  {scenario?.name}
                 </span>
-                {timetable && (
+                {trainScheduleV2Activated ? (
                   <span className="scenario-explorator-card-head-scenario-traincount">
-                    {validTrainCount(timetable.train_schedule_summaries)}
+                    {timetableV2 && v2TrainCount(timetableV2.train_ids)}
                     <MdTrain />
                   </span>
+                ) : (
+                  timetable && (
+                    <span className="scenario-explorator-card-head-scenario-traincount">
+                      {validTrainCount(timetable.train_schedule_summaries)}
+                      <MdTrain />
+                    </span>
+                  )
                 )}
               </div>
             </div>
             <div className="scenario-explorator-card-head-content-item">
               <img src={infraIcon} alt="infra icon" />
               <span className="scenario-explorator-card-head-legend">{t('infraLegend')}</span>
-              <div className="scenario-explorator-card-head-infra">
-                {scenarioDetails.infra_name}
-              </div>
+              <div className="scenario-explorator-card-head-infra">{scenario?.infra_name}</div>
```

Comment:
Now you know that the scenario is defined :+1: 
```suggestion
              <div className="scenario-explorator-card-head-infra">{scenario.infra_name}</div>
```